### DISCUSSION TITLE
Add MSWX to the archive API for low-latency high-resolution reanalysis data

### DISCUSSION THREAD

**Discussion by guidocioni** [open-meteo/open-meteo#454] [Ideas]
Probably you're already aware but I just wanted to point out that, in order to reduce the latency of data in the archive API, it would be an interesting idea to add the MSWX model (https://www.gloh2o.org/mswx/).

It is a high resolution (0.1 degrees) reanalysis available in near-real time which combines different sources of data, including ERA5. The advantages over ERA5 are not only the improved time latency and horizontal resolution, but also the way data can be easily downloaded without complicated queuing as in the CDS. In fact, they're available on Google Drive once registered, so you can easily set up a pipeline that downloads data fast without delay as soon as it is available: I use `rclone` for that.

I think it would be a nice addition to the models family. 

Note that only a bunch of surface variables are available.

**Comment by patrick-zippenfenig** [comment]
Thanks! I will write them and as for permission/access.

I am also working on different approaches to circumvent the ERA5 delay to realtime. I started to integrate the ECMWF IFS archive with 1 day delay. I am also considering to integrate some seamless mixing with other weather forecast domains. Ultimately, it is also an disk space and scalability issue, as all of this needs a lot of data :D

**Reply by patrick-zippenfenig** [reply]
Data looks ok on the first sight. However, it might be better to use GDAS directly.

![Screenshot 2023-09-28 at 18 50 49](https://github.com/open-meteo/open-meteo/assets/11670456/edbe99f0-5449-4aeb-80f5-c474fe728095)

**Reply by guidocioni** [reply]
I don't know exactly what is the real difference between GDAS and MSWX. I think they use it as input but still do some corrections. 
They have another product (MSWEP) which has only precipitation but uses only observations (satellites, gauges...) to reconstruct precipitation.
It may also worth a look :) 

Regarding IFS, unfortunately I still consider the resolution of open data (0.4 degrees) too coarse to be acceptable in 2023 :D I really hope ECMWF will make an effort to bring it at least at the same level of ERA5 (0.25 degrees):it doesn't really make sense nowadays to be that picky with the data.

**Reply by patrick-zippenfenig** [reply]
I've got access to the full resolution IFS and can redistribute **past data > 1-2 days** as CC-BY. Forecast is a different topic. Most of it is already implemented, it just needs finishing touches,  but right now I somehow got dragged into working on a binary serialisation format (FlatBuffers #448)

I have worked with MSWEP in the past with mixed results. Other satellite precipitation datasets like CHRIPS, CMORPH or IMERG could be a better fit.

**Reply by guidocioni** [reply]
ah ok, yes you're right, mswep sometimes has weird data. I've noticed it for storm Daniel over Libya that the amounts were way lower than CMORPH (which should be an input of mswep)...not really sure what's going on but probably they're compromising data quality by reducing latency.
Also when comparing to station data between era5-land and mswep I haven't seen much better skill scores
